### ISSUE TITLE
false positive in unexported-return

### ISSUE THREAD

**Issue by tamird** [mgechev/revive#5]
```
revive -formatter friendly github.com/cockroachdb/cockroach/pkg/...
  ⚠  unexported-return  exported func NewZeroNode returns unexported type sql.planNode, which can be annoying to use
  pkg/sql/zero.go:36:49
```
The code in question: https://github.com/cockroachdb/cockroach/blob/97c3622c2a6f4ec579ed93655b9d8a83732ac654/pkg/sql/zero.go#L36

It's true that the returned object is of an unexported type, but it is being returned as an interface, which is exported.

**Comment by mgechev** [comment]
Just like in golint, false  positives are possible in revive. The confidence of this warning is [0.8](https://github.com/mgechev/revive/blob/f74d3bd5eecded5f1b83c2e4b87f8cde22bd71d0/rule/unexported-return.go#L72). If you want to filter warnings with confidence below 1, you can use the config file.

**Comment by tamird** [comment]
Whoa! Closing this issue seems aggressive given that it should be possible to fix this false positive.

**Comment by mgechev** [comment]
This rule is directly ported from golint; I think it's reasonable to keep the current behavior with confidence level 0.8.

In case you'd be able to find some time to fix the false positive, I'd be more than happy to reopen the issue and introduce you to the codebase.
